core/interop/src/into_js_function_impls.rs

```diff
@@ -0,0 +1,180 @@
+        where
+            $($t: TryFromJsArgument + 'static,)*
+            R: TryIntoJsResult,
+            T: FnMut($($t,)*) -> R + 'static
+        {}
+
+        impl<$($t,)* R, T> IntoJsFunctionSealed<($($t,)* ContextArgToken), R> for T
+        where
+            $($t: TryFromJsArgument + 'static,)*
+            R: TryIntoJsResult,
+            T: FnMut($($t,)* &mut Context) -> R + 'static
+        {}
+
+        impl<$($t,)* R, T> UnsafeIntoJsFunction<($($t,)*), R> for T
+        where
+            $($t: TryFromJsArgument + 'static,)*
+            R: TryIntoJsResult,
+            T: FnMut($($t,)*) -> R + 'static,
+        {
+            #[allow(unused_variables)]
+            unsafe fn into_js_function_unsafe(self, _context: &mut Context) -> NativeFunction {
+                let s = RefCell::new(self);
+                unsafe {
+                    NativeFunction::from_closure(move |this, args, ctx| {
+                        let rest = args;
+                        $(
+                            let ($id, rest) = $t::try_from_js_argument(this, rest, ctx)?;
+                        )*
+                        match s.try_borrow_mut() {
+                            Ok(mut r) => r( $($id),* ).try_into_js_result(ctx),
+                            Err(_) => {
+                                Err(JsError::from_opaque(js_string!("recursive calls to this function not supported").into()))
+                            }
+                        }
+                    })
+                }
+            }
+        }
+
+        impl<$($t,)* R, T> UnsafeIntoJsFunction<($($t,)* ContextArgToken), R> for T
+        where
+            $($t: TryFromJsArgument + 'static,)*
+            R: TryIntoJsResult,
+            T: FnMut($($t,)* &mut Context) -> R + 'static,
+        {
+            #[allow(unused_variables)]
+            unsafe fn into_js_function_unsafe(self, _context: &mut Context) -> NativeFunction {
+                let s = RefCell::new(self);
+                unsafe {
+                    NativeFunction::from_closure(move |this, args, ctx| {
+                        let rest = args;
+                        $(
+                            let ($id, rest) = $t::try_from_js_argument(this, rest, ctx)?;
+                        )*
+                        let r = s.borrow_mut()( $($id,)* ctx);
+                        r.try_into_js_result(ctx)
+                    })
+                }
+            }
+        }
+
+        // Safe versions for `Fn(..) -> ...`.
+        impl<$($t,)* R, T> IntoJsFunctionCopied<($($t,)*), R> for T
+        where
+            $($t: TryFromJsArgument + 'static,)*
+            R: TryIntoJsResult,
+            T: Fn($($t,)*) -> R + 'static + Copy,
+        {
+            #[allow(unused_variables)]
+            fn into_js_function_copied(self, _context: &mut Context) -> NativeFunction {
+                let s = self;
+                unsafe {
+                    NativeFunction::from_closure(move |this, args, ctx| {
```

Comment:
In this case, I think you can use `from_copy_closure` instead.

Reply:
I seemed to have missed this. Done.